source/framework/vx_reference.c

```diff
@@ -2168,3 +2174,306 @@ vx_status tivxReferenceExportHandle(const vx_reference ref, void *addr[], uint32
+{
+    vx_user_data_object retref = NULL;
+    vx_status local_status;
+    vx_status *statusptr = (NULL != status) ? status : &local_status;
+
+    *statusptr = (vx_status)VX_SUCCESS;
+    if ( (vx_bool)vx_false_e == ownIsValidReference(ref))
+    {
+        *statusptr = (vx_status)VX_ERROR_INVALID_REFERENCE;
+        retref = (vx_user_data_object)ownGetErrorObject(ownGetContext(), (vx_status)VX_ERROR_INVALID_REFERENCE);
+    }
+    else if (((vx_bool)vx_true_e == ref->is_virtual) &&
+            ((vx_bool)vx_false_e == ref->is_accessible))
+    {
+        /* Can't get supplementary data from an inaccessible object*/
+        VX_PRINT(VX_ZONE_WARNING, "Attempt to get supplementary data from virtual object\n");
+        *statusptr = (vx_status)VX_ERROR_OPTIMIZED_AWAY;
+        retref = (vx_user_data_object)ownGetErrorObject(ref->context, (vx_status)VX_ERROR_OPTIMIZED_AWAY);
+    }
+    else if ((vx_enum)VX_TYPE_CONTEXT == ref->type)
+    {
+        /* Specifically avoid issues later */
+        *statusptr = (vx_status)VX_ERROR_NOT_SUPPORTED;
+        retref = (vx_user_data_object)ownGetErrorObject((vx_context)ref, (vx_status)VX_ERROR_NOT_SUPPORTED);
+    }
+    else if (NULL == ref->obj_desc)
+    {
+        /* objects without an object descriptor are not supported */
+        *statusptr = (vx_status)VX_ERROR_NOT_SUPPORTED;
+        retref = (vx_user_data_object)ownGetErrorObject(ref->context, (vx_status)VX_ERROR_NOT_SUPPORTED);
+    }
+    else
+    {
+        vx_reference new_ref = ref;
+        /* Look for parent data if this not in the kernel function or it is an input */
+        if (((vx_bool)vx_false_e == tivxFlagIsBitSet(ref->obj_desc->flags, TIVX_REF_FLAG_IS_IN_KERNEL)) ||
+            ((vx_bool)vx_true_e == tivxFlagIsBitSet(ref->obj_desc->flags, TIVX_REF_FLAG_IS_INPUT)))
+            while (NULL == new_ref->supplementary_data)
+            {
+                vx_reference parent = NULL;
+                if ((vx_enum)VX_TYPE_IMAGE == new_ref->type)
+                {
+                    parent = (vx_reference)((vx_image)new_ref)->parent;
+                }
+                if (NULL != parent)
+                {
+                    new_ref = parent;
+                }
+                else
+                {
+                    break;
+                }
+            };
+        if (NULL == new_ref->supplementary_data)
+        {
+            *statusptr = (vx_status)VX_FAILURE;
+            retref = (vx_user_data_object)ownGetErrorObject(ref->context, (vx_status)VX_FAILURE);
+        }
+        else if ((NULL != type_name) &&
+                 (0 != tivx_obj_desc_strncmp(((tivx_obj_desc_user_data_object_t *)new_ref->supplementary_data->base.obj_desc)->type_name, (volatile void *)type_name, VX_MAX_REFERENCE_NAME))
+                )
+        {
+            VX_PRINT(VX_ZONE_ERROR, "Type mismatch for supplementary data. Expected user type \"%s\"; got \"%s\"\n", type_name,
```

Comment:
@raphacan , this print is causing a seg fault on the board when running ... ran out of time today to narrow it down further.  We are out until Monday.  For now, we locally commented it out to get the coverage information.

Reply:
We have root caused this issue and pushed a fix to this branch.  Will mark as resolved accordingly